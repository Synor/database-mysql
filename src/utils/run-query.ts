type Connection = import('mysql').Connection

export type QueryValue = boolean | number | string | Date

export async function runQuery<T = any>(
  connection: Connection,
  query: string,
  values: QueryValue[] = []
): Promise<T> {
  return new Promise((resolve, reject) => {
    connection.query(query, values, (err, results) => {
      if (err) {
        reject(err)
        return
      }

      resolve(results)
    })
  })
}
