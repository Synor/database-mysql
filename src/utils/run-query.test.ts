import { createConnection } from 'mysql'
import { runQuery } from './run-query'

type Connection = import('mysql').Connection

const uri = 'mysql://root:root@127.0.0.1:3306/synor'

describe('utils:runQuery', () => {
  let connection: Connection

  beforeAll(() => {
    connection = createConnection(uri)
  })

  afterAll(() => {
    connection.destroy()
  })

  test('can execute query', async () => {
    await expect(runQuery(connection, 'SELECT 1;')).resolves.toMatchSnapshot()
  })

  test('can substitute values in query', async () => {
    await expect(
      runQuery(connection, 'SELECT ?;', [1])
    ).resolves.toMatchSnapshot()
  })

  test('throws if malformed query', async () => {
    await expect(runQuery(connection, 'SELEC 1;')).rejects.toThrow()
  })
})
