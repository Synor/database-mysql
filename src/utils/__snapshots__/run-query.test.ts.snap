// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils:runQuery can execute query 1`] = `
Array [
  RowDataPacket {
    "1": 1,
  },
]
`;

exports[`utils:runQuery can substitute values in query 1`] = `
Array [
  RowDataPacket {
    "1": 1,
  },
]
`;
